enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,279 @@
+---
+title: egress-ip-multi-nic
+authors:
+  - "@martinkennelly"
+reviewers:
+  - "@trozet"
+  - "@kyrtapz"
+  - "@flavio-fernandes"
+  - "@maiqueb"
+approvers:
+  - "@trozet"
+  - "@knobunc"
+api-approvers:
+- "@trozet"
+creation-date: 2023-05-16
+last-updated: 2023-06-07
+status: implementable
+tracking-link:
+- https://issues.redhat.com/browse/SDN-1123
+---
+# Egress IP Multi-NIC
+## Summary
+For OpenShift release version 4.13 or less, users may utilize egress IP to allow source NAT when packets leave OVN managed network
+by specifying the source NAT IP within an `EgressIP` kubernetes custom resource definition.
+This enhancement will add egress IP support for networks that exists on NICs that aren't managed by OVN and hence forth called
+non-ovn managed networks.
+
+## Motivation
+Users may not wish to route workload traffic over OVN managed network and instead prefer to use a non-ovn managed network.
+With this enhancement, we can support segregation of workload traffic over multiple NICs while allowing a configurable source IP address.
+
+### Goals
+- Support egress IP for non-ovn managed networks
+
+### Non-Goals
+- Support egress IP on non-default networks controllers aka secondary network controllers
+- Support cloud egress IP assignment for non-ovn managed interfaces using cloud network config controller
+  - This maybe easy to do eventually, but it will require a lot of extra testing, so excluding it for now, and it's
+    a stretch goal.
+- Support egress IP on secondary bridges managed by OVN aka br-ex1
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+Users may provide not only an ovn managed network but also additional non-ovn managed networks that they want to utilize and support egress IP.
+Cluster manager will continue to provision `EgressIP` to nodes. Currently, it only assigns them to nodes which have a
+well-known label egress assignable `k8s.ovn.org/egress-assignable`.
+
+This enhancement is proposing enhancing cluster manager if the egress IP is within a subnet of a known non-primary network, then it
+will also consider which node hosts this secondary network.
+
+Cluster manager will therefore need to understand all available networks within the cluster and this enhancement proposes
+to add all non-ovn managed interfaces addresses and subnet mask to existing annotation `k8s.ovn.org/host-addresses` which is
+added to all node custom resource objects.
+
+Egress IP must not equal an address already assigned to an interface, and we can use the aforementioned `k8s.ovn.org/host-addresses`
+annotation to detect this scenario. The addresses that can be added into this annotation are filtered by the following criteria:
+- Link must be up
+- Link must not be loopback, type openvswitch or bridge
+- Link must not have a master index (aka bridge)
+- Link must not have an associated parent index (aka slave devices)
```

Comment:
I'd remove this master / slave terminology for something more inclusive.

Reply:
Done. I cannot remove 'master' because thats what netlink calls it. I removed the slave reference in favour of child. Thanks!